testing/helpers.go

```diff
@@ -61,3 +65,62 @@ func BuildValidationRequest(object, settings easyjson.Marshaler) ([]byte, error)
 
 	return easyjson.Marshal(validationRequest)
 }
+
+// Extract PodSpec from high level objects. This method can be used to evaluate high level objects instead of just Pods.
+// For example, it can be used to reject Deployments or StatefulSets that violate a policy instead of the Pods created by them.
+// Objects supported are: Deployment, ReplicaSet, StatefulSet, DaemonSet, ReplicationController, Job, CronJob, Pod
+// It returns an error if the object is not one of those. If it is a supported object it returns the PodSpec if present, otherwise returns None.
+func ExtractPodSpecFromObject(object kubewarden_protocol.ValidationRequest) (corev1.PodSpec, error) {
+	switch object.Request.Kind.Kind {
```

Comment:
Nitpicking, I don't know if would be worth to be more specific when checking the inner object. Like by looking not just at `Kind`, but also at `Group`. Just to make sure this is `apps`, `Deployment`.  :thinking:

Reply:
I think using `Request.Kind.Kind` is enough, `kind` is fully qualified (`v1.Pods` for example).
See: https://github.com/kubewarden/policy-sdk-go/blob/fd7f77e3f84b1335f2532dc0af9537db9cd694d5/protocol/types.go#L50-L51

This means though that the switch arms need adjustment.

Reply:
Good point @viccuad. @KhaledEmaraDev can you please rewrite the switch to match what @viccuad proposed, please?